x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -288,6 +289,13 @@ else if (exp instanceof GroupingFunction) {
                                 if (h.dataType() == DATE) {
                                     intervalAsMillis = DateUtils.minDayInterval(intervalAsMillis);
                                 }
+
+                                // When the histogram in SQL is applied on TIME type instead of DATETIME, the interval
+                                // specified is truncated to maximum 1 day. If the interval specified is less
+                                // than 1 day, then the interval specified is used without any change.
+                                if (h.dataType() == TIME) {
```

Comment:
`else if`

Reply:
Cannot enter both ifs but will change to make it more clear.